solr/core/src/java/org/apache/solr/update/processor/TimeRoutedAliasUpdateProcessor.java

```diff
@@ -272,7 +349,8 @@ private String findTargetCollectionGivenTimestamp(Instant docTimestamp) {
         return entry.getValue(); //found it
       }
     }
-    return null; //not found
+    throw new SolrException(SolrException.ErrorCode.BAD_REQUEST,
+        "Doc " + id + " couldn't be routed with " + timeRoutedAlias.getRouteField() + "=" + docTimestamp);
```

Comment:
I think the error should say it's too old for the oldest collection

Reply:
That would be clearer, I agree